src/ert/config/summary_config.py

```diff
@@ -23,7 +23,11 @@
 class SummaryConfig(ResponseConfig):
     input_file: str
     keys: List[str]
-    refcase: Optional[Set[datetime]] = None
+    refcase: Optional[Union[Set[datetime], List[str]]] = None
```

Comment:
Prefer just using `Union` rather than mixing `Union` and `Optional`:
```suggestion
    refcase: Union[Set[datetime], List[str], None] = None
```